### PR TITLE
[DO NOT MERGE] swagger for doc and schema

### PR DESCRIPTION
**Pull request by DeepakRajendranMsft**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### Gist of this PR
- Removed `$expand query parameter` from applicable operations. (i.e. $expand scenario does not exist)
- Added constraint `"readOnly": true` to properties as applicable
- Added `required` properties to models as applicable
- Specs touched in this PR will be used for generating REST API docs and Resource schemas.
- Other/Original version of the specs will be used to generate SDKs (business as usual).
- Whenever a change is made to the version of specs for generating SDKs (or vice-versa) a corresponding PR will be issued by networking team to ensure that the version of specs for Docs and schema generation is updated. 

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by balajikris** [comment]
Hey @DeepakRajendranMsft - Sometime last week, this repository has changed the default branch to use the branch labeled `current`. I think `master` no longer accepts new PRs, nor do I have permissions to merge to this branch. Please open a PR against `current` branch.
If you have any questions or concerns, please ping the #swagger channel in slack.

**Comment by DeepakRajendranMsft** [comment]
@balajikris the idea here is not to merge. let me add you to an internal thread

**Comment by balajikris** [comment]
@DeepakRajendranMsft - thanks. feel free to tag anyone who have been helping you with this process to get their review. The bot randomly assigns folks to PRs that are to be merged into this repo. If someone already has more context than I do and if you'd like their review, feel free to tag them.

**Comment by DeepakRajendranMsft** [comment]
@amarzavery @salameer  please take a look

**Comment by DeepakRajendranMsft** [comment]
CI says swagger "/home/travis/build/Azure/azure-rest-api-specs/arm-network/2017-06-01/swagger/network.json" is an invalid JSON.

but jsontlint.com says its fine....

**Comment by balajikris** [comment]
regarding the above, i dont think its a syntax error. If you scrolled all the way to the end, it says what the error is:

~~~
 1) Azure swagger schema validation: /home/travis/build/Azure/azure-rest-api-specs/arm-network/2017-06-01/swagger/network.json should be a valid Swagger document.:
     swagger "/home/travis/build/Azure/azure-rest-api-specs/arm-network/2017-06-01/swagger/network.json" is an invalid JSON.
Error: Schema validation failed: [ { code: 'INVALID_TYPE',
    params: [ 'boolean', 'string' ],
    message: 'Expected type boolean but found type string',
    path: '#/definitions/SubResourceReadOnly/x-ms-azure-resource',
    schemaId: 'https://raw.githubusercontent.com/Azure/autorest/master/schema/swagger-extensions.json#' },
  { code: 'OBJECT_ADDITIONAL_PROPERTIES',
    params: [ [ 'readonly' ] ],
    message: 'Additional properties not allowed: readonly',
    path: '#/definitions/SubResourceReadOnly/properties/id',
    schemaId: 'https://raw.githubusercontent.com/Azure/autorest/master/schema/swagger-extensions.json#',
    description: 'A deterministic version of a JSON Schema object.' } ]
    at /home/travis/build/Azure/azure-rest-api-specs/test/syntax.js:52:17
    at /home/travis/build/Azure/azure-rest-api-specs/test/util/utils.js:56:14
    at tryToString (fs.js:456:3)
    at FSReqWrap.readFileAfterClose [as oncomplete] (fs.js:443:12)
  Error: swagger "/home/travis/build/Azure/azure-rest-api-specs/arm-network/2017-06-01/swagger/network.json" is an invalid JSON.
  Error: Schema validation failed: [ { code: 'INVALID_TYPE',
      params: [ 'boolean', 'string' ],
      message: 'Expected type boolean but found type string',
      path: '#/definitions/SubResourceReadOnly/x-ms-azure-resource',
      schemaId: 'https://raw.githubusercontent.com/Azure/autorest/master/schema/swagger-extensions.json#' },
    { code: 'OBJECT_ADDITIONAL_PROPERTIES',
      params: [ [ 'readonly' ] ],
      message: 'Additional properties not allowed: readonly',
      path: '#/definitions/SubResourceReadOnly/properties/id',
      schemaId: 'https://raw.githubusercontent.com/Azure/autorest/master/schema/swagger-extensions.json#',
      description: 'A deterministic version of a JSON Schema object.' } ]
      at test/syntax.js:52:17
      at test/util/utils.js:56:14
      at tryToString (fs.js:456:3)
      at FSReqWrap.readFileAfterClose [as oncomplete] (fs.js:443:12)
      at test/util/utils.js:58:15
      at tryToString (fs.js:456:3)
      at FSReqWrap.readFileAfterClose [as oncomplete] (fs.js:443:12)
npm ERR! Test failed.  See above for more details.
~~~

**Comment by DeepakRajendranMsft** [comment]
@balajikris  thanks for that. always learning new things :)

**Comment by balajikris** [comment]
still failing schema validation

**Comment by DeepakRajendranMsft** [comment]
@balajikris  works now

@amarzavery please take a look

**Comment by salameer** [comment]
@amarzavery please take a look

File: arm-network/2017-06-01/swagger/network.json [hunk 1]
```diff
@@ -122,7 +121,10 @@
           "description": "Resource tags."
         }
       },
-      "x-ms-azure-resource": true
+      "x-ms-azure-resource": true,
```
Comment (amarzavery) [inline arm-network/2017-06-01/swagger/network.json]: Resource.id should also be marked as `"readOnly": true`

**Comment by amarzavery** [comment]
Made one comment. Everything else LGTM.

@johanste @salameer @DeepakRajendranMsft @markjbrown 

Gist of this PR
- Removed $expand query parameter from applicable operations. (i.e. $expand scenario does not exist)
- Added constraint `"readOnly": true` to properties as applicable
- Added `required` properties to models as applicable
- Specs touched in this PR will be used for generating REST API docs and Resource schemas.
- Other/Original version of the specs will be used to generate SDKs (business as usual).
- Whenever a change is made to the version of specs for generating SDKs (or vice-versa) a corresponding PR will be issued by networking team to ensure that the version of specs for Docs and schema generation is updated.

File: arm-network/2017-06-01/swagger/network.json (hunk 1 above)
Comment (DeepakRajendranMsft) [inline arm-network/2017-06-01/swagger/network.json]: done

**Comment by salameer** [comment]
Captain @DeepakRajendranMsft 

Anymore review related items remaining on this?

**Comment by fearthecowboy** [comment]
@DeepakRajendranMsft -- we've gone thru a reorganization in this repository -- can I get you to fork a new copy of this repository and base your work off the `current` branch, and then issue a new PR?

Thanks.

**Comment by DeepakRajendranMsft** [comment]
closing this since we got this reviewed

**Pull request closed without merging**